core/src/main/java/org/elasticsearch/action/RealtimeRequest.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action;
+
+/**
+ * Indicates that a request can execute in realtime (reads from the translog).
+ * All {@link ActionRequest} that are realtime should implement this interface.
+ */
+public interface RealtimeRequest {
+
+    /**
+     * @param realtime Controls whether this request should be realtime by reading from the translog.
+     */
+    <R extends RealtimeRequest> R realtime(Boolean realtime);
```

Comment:
hmmm... in all cases where null is being normalised to true. So instead documenting what `null` mean we should simplify and change `realtime` field in all places to be a primitive boolean and default to `true`.